tests/TestHelpers/HttpRequestHelper.php

```diff
@@ -216,17 +216,17 @@ public static function sendRequest(
 				$client
 			);
 
-			if ($response->getStatusCode() >= 400 && $response->getStatusCode() !== self::HTTP_TOO_EARLY) {
+			if ($response->getStatusCode() >= 400 && $response->getStatusCode() !== self::HTTP_TOO_EARLY && $response->getStatusCode() !== self::HTTP_CONFLICT) {
 				$sendExceptionHappened = true;
 			}
 
 			if ($debugResponses) {
 				self::debugResponse($response);
 			}
 			$sendCount = $sendCount + 1;
-			$loopAgain = !$sendExceptionHappened && ($response->getStatusCode() === self::HTTP_TOO_EARLY) && ($sendCount <= $sendRetryLimit);
+			$loopAgain = !$sendExceptionHappened && ($response->getStatusCode() === self::HTTP_TOO_EARLY || $response->getStatusCode() === self::HTTP_CONFLICT) && $sendCount <= $sendRetryLimit;
```

Comment:
currently this function has 47 usages but we only need to loop this when we get `409` while making a folder so i think we should destructure this into 
```suggestion
                        $sendExceptionConflictHappened = false;
			if ($response->getStatusCode() >= 400 && $response->getStatusCode() !== self::HTTP_TOO_EARLY && $response->getStatusCode() !== self::HTTP_CONFLICT) {
				$sendExceptionHappened = true;
			}

            if ($response->getStatusCode() === self::HTTP_CONFLICT && substr($url,44)==="https://localhost:9200/remote.php/dav/files/") {
                $sendExceptionConflictHappened = true;
            }

			if ($debugResponses) {
				self::debugResponse($response);
			}
			$sendCount = $sendCount + 1;
			$loopAgain = !$sendExceptionHappened && ($response->getStatusCode() === self::HTTP_TOO_EARLY || $response->getStatusCode() === self::HTTP_CONFLICT) && $sendCount <= $sendRetryLimit;
            $loopAgainConflict = !$sendExceptionConflictHappened && $sendCount <= $sendRetryLimit;
			if ($loopAgain || $loopAgainConflict) {
				// we need to repeat the send request, because we got HTTP_TOO_EARLY or HTTP_CONFLICT
				// wait 1 second before sending again, to give the server some time
				// to finish whatever post-processing it might be doing.
				\sleep(1);
```